### PR TITLE
Remove "dvrSeekLimit" from Controlbar and Timeslider

### PR DESCRIPTION
**Pull request by DanFerrer**
### This PR will...

Remove "dvrSeekLimit" from the duration and position in the UI only.

### Why is this Pull Request needed?

The seek bar would factor in the full duration of the stream so it would appear you were allowed to seek further than you could, but we don't allow you to seek further than the live edge minus the seek limit

### Are there any Pull Requests open in other repos which need to be merged with this?

https://github.com/jwplayer/jwplayer-commercial/pull/4949

#### Addresses Issue(s):
JW8-1359

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,14380 @@
+{
```
Comment (robwalch) [inline package-lock.json]: Do not add "package-lock.json" to the project. It was removed intentionally. We use yarn.lock.

File: src/js/providers/default.js [hunk 2]
```diff
@@ -63,6 +63,8 @@ const DefaultProvider = {
     getCurrentAudioTrack: noop,
     setCurrentAudioTrack: noop,
 
+    getSeekRange: noop,
```
Comment (robwalch) [inline src/js/providers/default.js]: `getSeekRange` must be implemented in html5.

Comment (robwalch) [reply src/js/providers/default.js]: If this method returns nothing, what will happen for code that expects a range returned?

File: src/js/view/controls/components/timeslider.js [hunk 3]
```diff
@@ -157,7 +141,9 @@ class TimeSlider extends Slider {
         var pct = 0;
         if (duration) {
             if (this.streamType === 'DVR') {
-                pct = (duration - position) / duration * 100;
+                var seekRange = this._model.get('seekRange');
```
Comment (robwalch) [inline src/js/view/controls/components/timeslider.js]: Not all providers set this property.

Use `const` not `var`.

File: src/js/providers/default.js (hunk 2 above)
Comment (robwalch) [reply src/js/providers/default.js]: By default this should return `{ start: 0, end: duration } ` for backwards compatibility.

Comment (robwalch) [reply src/js/providers/default.js]: @DanFerrer @johnBartos ^^

**Comment by DanFerrer** [comment]
@robwalch It was removed because its was being factored into the end of the seek range. So we substract from the seekable end of the video to make sure the viewer is behind enough

**Comment by robwalch** [comment]
@DanFerrer subtract what?

**Comment by DanFerrer** [comment]
test this please

File: src/js/providers/html5.js [hunk 4]
```diff
@@ -282,6 +282,24 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
...
+            end: 0
+        };
+
+        const seekable = this.video.seekable;
+
+        if (seekable.length) {
+            const maxSeekableEnd = Math.max(seekable.end(0), seekable.end(seekable.length - 1));
+
+            seekRange.end = maxSeekableEnd;
+            seekRange.start = Math.max(0, seekRange.end - this.video.duration);
```
Comment (robwalch) [inline src/js/providers/html5.js]: Why not use `seekable.start(0)`?

or 

`Math.min(seekable.start(0), seekable.start(seekable.length - 1));`

File: src/js/providers/html5.js [hunk 5]
```diff
@@ -282,6 +282,24 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
         return duration;
     };
 
+    _this.getSeekRange = function() {
+        const seekRange = {
+            start: 0,
+            end: 0
```
Comment (robwalch) [inline src/js/providers/html5.js]: default end should be `duration`. You could invoke `Default. getSeekRange()`

**Comment by robwalch** [comment]
test this please

**Pull request merged**